src/Xtools/ArticleInfoRepository.php

```diff
@@ -36,21 +41,51 @@ public function getBotData(Page $page)
 
     /**
      * Get prior deletions, page moves, and protections to the page.
-     * @param  Page page
+     * @param Page $page
+     * @param false|int $start
+     * @param false|int $end
      * @return string[] each entry with keys 'log_action', 'log_type' and 'timestamp'.
      */
-    public function getLogEvents(Page $page)
+    public function getLogEvents(Page $page, $start, $end)
     {
         $loggingTable = $page->getProject()->getTableName('logging', 'logindex');
+
+        $datesConditions = self::createDatesConditions($start, $end, '', 'log_timestamp');
+
         $sql = "SELECT log_action, log_type, log_timestamp AS 'timestamp'
                 FROM $loggingTable
                 WHERE log_namespace = '" . $page->getNamespace() . "'
-                AND log_title = :title AND log_timestamp > 1
+                AND log_title = :title AND log_timestamp > 1 $datesConditions
                 AND log_type IN ('delete', 'move', 'protect', 'stable')";
         $title = str_replace(' ', '_', $page->getTitle());
         $resultQuery = $this->getProjectsConnection()->prepare($sql);
         $resultQuery->bindParam(':title', $title);
         $resultQuery->execute();
         return $resultQuery->fetchAll();
     }
+
+    /**
+     * Creates WHERE conditions with date range to be put in query.
+     *
+     * @param false|int $start
+     * @param false|int $end
+     * @param string $tableAlias Alias of table FOLLOWED BY DOT.
+     * @param string $field
+     * @return string
+     */
+    public static function createDatesConditions($start, $end, $tableAlias = '', $field = 'rev_timestamp')
```

Comment:
What if we moved this to `Repository`? Then it's accessible in PageRepository, ArticleInfoRepository, etc., and it doesn't have to be static.

Reply:
I second that